### ISSUE TITLE
Change the `modifyValueOnWheel` default behaviour to act only when the element is focused

### ISSUE THREAD

**Issue by wc-matteo** [autoNumeric/autoNumeric#456]
With an option to disable the focus constraint (making it as it is now).

**Comment by AlexandreBonneau** [comment]
Considering the comments in #393, I'm adding a new `wheelOn` option that can accept either the `'focus'` or `'hover'` value.

Using the `Shift` key while using the mouse wheel will **invert** the object behavior (if it was supposed to increment/decrement, then it does nothing, and vice versa).

@wc-matteo What do you think about this wheel event management?:
```js
// 6 cases are possibles
if (this.settings.modifyValueOnWheel) {
    switch (this.settings.wheelOn) {
        case AutoNumeric.options.wheelOn.focus:
            if (this.isFocused) {
                if (e.shiftKey) {
                    // Deactivate the wheel
                } else {
                    // Activate the wheel
                }
            } else {
                if (e.shiftKey) {
                    // Activate the wheel
                } else {
                    // Deactivate the wheel
                }
            }

            break;
        case AutoNumeric.options.wheelOn.hover:
            if (e.shiftKey) {
                // Deactivate the wheel
            } else {
                // Activate the wheel
            }

            break;
        default :
            AutoNumericHelper.throwError(`Unknown wheelOn option`);
    }
}
```

**Comment by AlexandreBonneau** [comment]
This is fixed in `v4.1.0-beta.16` with the new `wheelOn` option.
